software/duet_config_files/duet2/rrf3/sys/homez.g

```diff
@@ -1,4 +1,14 @@
 ; Home Z Axis
+
+if !move.axes[3].homed
+  M291 R"Cannot Home Z" P"U axis must be homed before Z to prevent damage to tool. Press OK to home U or Cancel to abort" S3
+  G28 U
+
+if state.currentTool != -1
+  M84 U
+  M291 R"Cannot Home Z" P"Tool must be deselected before homing. U has been unlocked, please manually dock tool and press OK to continue or Cancel to abort" S3
+  G28 U
+
```

Comment:
Technically X and Y need to be homed for Z to be homeable. Can you confirm that RRF3 prevents you from homing z without first homing X and Y? If that's the case, let's add the comment that

```RRF3 checks that X and Y are homed before running Z's 3 point bed leveling routine.```

Reply:
Comment added.
It does. A conditional check inside the program won't work, so I can't auto prompt the user to home X&Y.